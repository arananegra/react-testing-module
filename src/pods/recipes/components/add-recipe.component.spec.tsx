import * as React from 'react';
import { render } from '@testing-library/react';
import { AddRecipe } from './add-recipe.component';

describe('Add recipe tests', () => {
  it('should render a happy path button with simple props', () => {
    // Act
    const { getByTestId, asFragment } = render(<AddRecipe/>);
    const addRecipeElement = getByTestId('mui-button') as HTMLButtonElement;
    //Assert

    expect(addRecipeElement).not.toBeUndefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
