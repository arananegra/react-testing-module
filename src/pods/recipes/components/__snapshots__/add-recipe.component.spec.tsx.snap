// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add recipe tests should render a happy path button with simple props 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="mui-button"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
